src/parser/ParsedQuery.h

```diff
@@ -337,21 +337,39 @@ class ParsedQuery {
 
     // Add a variable, that was found in the query body. The added variables
     // will only be used if `isAsterisk` is true.
-    void registerVariableVisibleInQueryBody(const string& variable) {
+    void aux_registerVariableVisibleInQueryBody(const string& variable) {
       if (!(std::find(_variablesFromQueryBody.begin(),
                       _variablesFromQueryBody.end(),
                       variable) != _variablesFromQueryBody.end())) {
         _variablesFromQueryBody.emplace_back(variable);
       }
     }
 
-    // Gets the variables which registerVariableVisibleInQueryBody` was
+    // Get the variables for which `registerVariableVisibleInQueryBody` was
     // previously called.
     // The result contains no duplicates and is ordered by the first appearance
     // in the query body.
     [[nodiscard]] const auto& orderedVariablesFromQueryBody() const {
       return _variablesFromQueryBody;
     }
+
+    // Get the variables accordingly to established Selector:
+    // Select All (Select '*')
+    // or
+    // explicit variables selection (Select 'var_1' ... 'var_n')
+    [[nodiscard]] auto& getAccordinglySelectVariables() {
+      return isAsterisk() ? _variablesFromQueryBody
+                          : std::get<std::vector<string>>(_varsOrAsterisk);
+    }
+
+    // Get the variables accordingly to established Selector:
+    // Select All (Select '*')
+    // or
+    // explicit variables selection (Select 'var_1' ... 'var_n')
+    [[nodiscard]] const auto& getAccordinglySelectVariables() const {
+      return isAsterisk() ? _variablesFromQueryBody
+                          : std::get<std::vector<string>>(_varsOrAsterisk);
+    }
```

Comment:
1. I would just call it `getSelectedVariable`
2. I would only leave the `const` overload in, because you shouldn't manually alter these variables.